x-pack/plugins/siem/public/store/timeline/helpers.ts

```diff
@@ -252,8 +271,15 @@ const addProviderToTimeline = (
   timelineById: TimelineById
 ): TimelineById => {
   const alreadyExistsAtIndex = timeline.dataProviders.findIndex(p => p.id === provider.id);
+
+  if (alreadyExistsAtIndex > -1 && !isEmpty(timeline.dataProviders[alreadyExistsAtIndex].and)) {
+    provider.id = `${provider.id}-${
+      timeline.dataProviders.filter(p => p.id === provider.id).length
```

Comment:
In testing I was able to find a crash with duplicate keys, but I haven't been able to reproduce it. I had a pretty complicated query going with multiple dupes and was removing/adding terms back, so definitely a corner case. If I manage to find a way to repro I'll open an issue for tracking.

Reply:
let's keep an eye on it we work on other bugs